# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

##Hipótesis Nula:
No hay diferencia significativa en la pérdida de peso promedio de las personas que siguieron la dieta 1, y las personas que siguieron la dieta 2

##Hipótesis Alternativa:
Hay una diferencia significativa en la pérdida de peso promedio entre las personas que hicieron la dieta 1 y las personas que siguieron la dieta 2  

In [2]:
# TODO
from scipy import stats

Diet1 = [2.0,2.5,3.0,2.8,2.3,2.7,2.5]
Diet2 = [3.0,3.2,3.1,2.9,2.8,3.0,3.2]

t_value, p_value = stats.ttest_ind(Diet1, Diet2)

print(f"t-value: {t_value}")
print(f"p-value: {p_value}")

t-value: -3.5383407969933938
p-value: 0.004083270191713912


##Conclusión
Después de realizar la prueba T-Student se obtuvieron los siguientes valores:
 - t: -3.5383
 - p: 0.00408

 El valor de significancia se establece como 0.05, y ya que p obtuvo un valor menor a ese, se rechaza la hipótesis nula y se establece que hay una diferencia de peso significativa entre el peso de las personas que siguieron dietas distintas

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

Hipótesis nula: las medias de los tres grupos de fertilizantes es la misma
Hipótesis alterna: al menos una de las medias de los tres grupos es distinta

In [2]:
# TODO
import scipy.stats as stats

fert1 = [20, 21, 20, 19, 20]
fert2 = [22, 21, 23, 22, 21]
fert3 = [24, 23, 22, 23, 24]

# ANOVA test
f_value, p_value = stats.f_oneway(fert1, fert2, fert3)

print(f"f-value: {f_value}")
print(f"p-value: {p_value}")

f-value: 20.315789473684188
p-value: 0.00014047824793190475


Se establece el valor de significancia en 0.05, y con esto se rechaza la hipótesis nula, con lo cual concluimos que al menos uno de los 3 fertilizantes hace una diferencia significativa.

¿Cómo se puede determinar cuál es el mejor fertilizante?

Existen distintos tipos de análisis post-hoc que pueden ayudar a determinar la diferencia entre los grupos, como el Nemenyi Test y Tukey HSD Test

In [7]:
import scikit_posthocs as sp
import numpy as np
from statsmodels.stats.multicomp import pairwise_tukeyhsd

data1 = np.array([fert1, fert2, fert3])
#perform Nemenyi post-hoc test
print("Nemenyi Results: ")
print(sp.posthoc_nemenyi_friedman(data1.T))
print("\n")

print("Tukey Results: ")
tukey_results = pairwise_tukeyhsd(np.concatenate([fert1, fert2, fert3]),
                                  np.concatenate([['A'] * len(fert1), ['B'] * len(fert2), ['C'] * len(fert3)]))

print(tukey_results)

Nemenyi Results: 
          0         1         2
0  1.000000  0.253784  0.019679
1  0.253784  1.000000  0.509841
2  0.019679  0.509841  1.000000


Tukey Results: 
Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj  lower  upper  reject
--------------------------------------------------
     A      B      1.8 0.0099 0.4572 3.1428   True
     A      C      3.2 0.0001 1.8572 4.5428   True
     B      C      1.4 0.0409 0.0572 2.7428   True
--------------------------------------------------


Con los resultados obtenidos, se concluye que hay una diferencia significativa dependiendo del uso de distintos fertilizantes. Los resultados nos indican que el fertilizante 2 no muestra una diferencia significativa respecto a el fertilizante 1 y 3, sin embargo, al comparar los fertilizantes 1 y 3, se infiere que el fertilizante 3 tiene un mejor desempeño en comparación con el fertilizante 1, por lo que el 3 es el mejor